compiler/ast.nim

```diff
@@ -314,6 +314,7 @@ type
                       # an infinite loop, this flag is used as a sentinel to stop it.
     sfVirtual         # proc is a C++ virtual function
     sfByCopy          # param is marked as pass bycopy
+    sfMember          # proc is a C++ member of a type
```

Comment:
Can't `sfVirtual` be subsumed by `sfMember`? The RFC hints that it can.

Reply:
The syntax will get even more complex (assuming you are referring to add the keyword to the virtual syntax)

Reply:
That's what I meant yes. Please show me the resulting patterns: "now" vs "without virtual support"

Reply:
A pretty basic example would be like this:
```nim
proc foo(x: Doo): int {. virtual .} = 1
proc foo(x: Doo): int {. member:"virtual $1('1 #1) -> '0" .} = 1
```
Notice it wouldnt work as it is as I would need to modify the parser to detect `virtual`. 

What I would do is to have as part of the std a `cpputils.nim` module. So we can have `virtual` defined as a macro alongside `super`, `constcpp` (will require to expand `codegenDecl` to work on params), etc. `virtual` could also support the syntax as an argument so things keep working as they are working today.

Then `member` could be a low level interface where `cpputils.nim` can built on top of and the user doesnt need to use the syntax but the macros (we also have `byref` implemented already). If you are not willing to accept that, I wouldnt separate `virtual` from `member` as it makes it more complex.